SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerImpl.java

```diff
@@ -327,10 +358,65 @@ private void callWithStateLock(Runnable function, String name, boolean ignoreIfN
     }
   }
 
+  public void reconnectMesos() {
+    callWithStateLock(() -> {
+      state.setMesosSchedulerState(MesosSchedulerState.PAUSED_FOR_MESOS_RECONNECT);
+      mesosSchedulerClient.close();
```

Comment:
Can this throw an exception?

Reply:
Yes, but that will bubble to our uncaught exception handling for anything in this class (which will end up causing an abort). I could work in some more handling for weirdness stopping the mesos connection, but to start with I think it's a good improvement to be able to retry it, and still abort if something goes wrong in that retrying flow

Reply:
If the exception gets handled higher then it should be fine. I was mostly concerned about it getting stuck in a paused state indefinitely.